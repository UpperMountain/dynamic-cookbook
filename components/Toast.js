import React from "react";
import {
  StyleSheet,
  Modal,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
  Alert
} from "react-native";
import RoundedButton from "./RoundedButton";
var s = require("./Styles");

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    bottom: 0,
    height: 320,
    width: 304,
    borderTopLeftRadius: 39,
    borderTopRightRadius: 39,
    backgroundColor: "white",
    paddingTop: 42,
    paddingLeft: 35,
<<<<<<< HEAD
    paddingRight: 35,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5
=======
    paddingRight: 35
>>>>>>> general progress
  },
  modal: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  }
});

class Toast extends React.Component {
  constructor(props) {
    super();
    this.state = {
      open: false
    };
  }

  toggle = () => this.setState({ open: !this.state.open });

  render() {
    return (
      <View>
        <RoundedButton clear text="Test Button" onPress={this.toggle} />
        <Modal
          animationType={"slide"}
          transparent={true}
          visible={this.state.open}
        >
          <TouchableOpacity
            activeOpacity={1}
            style={styles.modal}
            onPress={this.toggle}
          >
            <TouchableWithoutFeedback>
              <View style={[styles.container, s.shadow]}>
                <Text style={s.header}>{this.props.headerText}</Text>
                <Text style={s.body}>{this.props.bodyText}</Text>
              </View>
            </TouchableWithoutFeedback>
          </TouchableOpacity>
        </Modal>
      </View>
    );
  }
}
export default Toast;
