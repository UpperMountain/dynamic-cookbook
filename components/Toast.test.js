import React from "react";
import Toast from "./Toast";
import renderer from "react-test-renderer";

it("renders without exploding", () => {
  const cmp = renderer.create(<Toast />);
});

it("passes snapshot tests", () => {
  const cmp = renderer.create(<Toast />);
  expect(cmp.toJSON()).toMatchSnapshot();
});
