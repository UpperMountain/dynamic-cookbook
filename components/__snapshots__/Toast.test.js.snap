// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot tests 1`] = `
<View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "white",
          "borderRadius": 72,
          "borderWidth": 3,
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "center",
          "opacity": 1,
          "width": 315,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flexDirection": "row",
            "fontFamily": "raleway-bold",
            "fontSize": 18,
            "lineHeight": 32,
          }
        }
      >
        Test Button
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderTopLeftRadius": 39,
              "borderTopRightRadius": 39,
              "bottom": 0,
              "height": 320,
              "paddingLeft": 35,
              "paddingRight": 35,
              "paddingTop": 42,
              "position": "absolute",
              "width": 304,
            },
            Object {
              "elevation": 8,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.27,
              "shadowRadius": 4.65,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "raleway-medium",
              "fontSize": 24,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "raleway",
              "fontSize": 14,
            }
          }
        />
      </View>
    </View>
  </Modal>
</View>
`;
